client/task-list/tasks/payments/razorpay.js

```diff
@@ -0,0 +1,175 @@
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+const INITIAL_CONFIG = {
+	key_id: '',
+	key_secret: '',
+};
+
+const validate = ( values ) => {
+	const errors = {};
+
+	if ( ! values.key_id ) {
+		errors.key_id = __( 'Please enter your Key ID', 'woocommerce-admin' );
+	}
+
+	if ( ! values.key_secret ) {
+		errors.key_secret = __(
+			'Please enter your Key Secret',
+			'woocommerce-admin'
+		);
+	}
+
+	return errors;
+};
+
+const updateSettings = async (
+	values,
+	createNotice,
+	markConfigured,
+	updateOptions
+) => {
+	const update = await updateOptions( {
+		woocommerce_razorpay_settings: {
+			key_id: values.key_id,
+			key_secret: values.key_secret,
+			enabled: 'yes',
+		},
+	} );
+
+	if ( update.success ) {
+		markConfigured( 'razorpay' );
+		createNotice(
+			'success',
+			__( 'Razorpay connected successfully', 'woocommerce-admin' )
+		);
+	} else {
+		createNotice(
+			'error',
+			__(
+				'There was a problem saving your payment settings',
+				'woocommerce-admin'
+			)
+		);
+	}
+};
+
+const renderConnectStep = ( {
+	createNotice,
+	isOptionsRequesting,
+	markConfigured,
+	updateOptions,
+} ) => {
+	const helpText = interpolateComponents( {
+		mixedString: __(
+			'Your key details can be obtained from your {{link}}Razorpay account{{/link}}',
+			'woocommerce-admin'
+		),
+		components: {
+			link: (
+				<Link
+					href="https://dashboard.razorpay.com/#/access/signin"
+					target="_blank"
+					type="external"
+				/>
+			),
+		},
+	} );
+
+	const onSubmit = ( values ) =>
+		updateSettings( values, createNotice, markConfigured, updateOptions );
+
+	return (
+		<Form
+			initialValues={ INITIAL_CONFIG }
+			onSubmitCallback={ onSubmit }
+			validate={ validate }
+		>
+			{ ( { getInputProps, handleSubmit } ) => {
+				return (
+					<>
+						<TextControl
+							label={ __( 'Key ID', 'woocommerce-admin' ) }
+							required
+							{ ...getInputProps( 'key_id' ) }
+						/>
+						<TextControl
+							label={ __( 'Key Secret', 'woocommerce-admin' ) }
+							required
+							{ ...getInputProps( 'key_secret' ) }
+						/>
+						<Button
+							isPrimary
+							isBusy={ isOptionsRequesting }
+							onClick={ handleSubmit }
+						>
+							{ __( 'Proceed', 'woocommerce-admin' ) }
+						</Button>
+
+						<p>{ helpText }</p>
+					</>
+				);
+			} }
+		</Form>
+	);
+};
+
+export const Razorpay = ( {
+	createNotice,
+	installStep,
+	isOptionsRequesting,
+	markConfigured,
+	updateOptions,
+} ) => {
+	return (
+		<Stepper
+			isVertical
+			isPending={ ! installStep.isComplete || isOptionsRequesting }
+			currentStep={ installStep.isComplete ? 'connect' : 'install' }
+			steps={ [
+				installStep,
+				{
+					key: 'connect',
+					label: __(
+						'Connect your Razorpay account',
+						'woocommerce-admin'
+					),
+					content: renderConnectStep( {
+						createNotice,
+						isOptionsRequesting,
+						markConfigured,
+						updateOptions,
+					} ),
+				},
+			] }
+		/>
+	);
+};
+
+export default compose(
```

Comment:
Not a blocker, but do we want to make use of `useSelect` and `useDispatch` within the Razorpay component instead of using `compose`?
I assume that eventually `withSelect` and `withDispatch` will get depreciated.

Reply:
Sure, I'll refactor that - this is a result of copy-pasta 🍝  😄

Reply:
Done in cd1d7b0.

Reply:
Nice!! Although I would suggest changing [this line](https://github.com/woocommerce/woocommerce-admin/commit/cd1d7b0e4dfaf3d10c73f1067daa06d5a1d20e89#diff-4b28b4c6e4e9c42defbabdcdb407722794cbe325b7bfe293d3182aa5c43fca80R157-R159) to this (similar to how you where using withSelect previously):
```
const isOptionsUpdating = useSelect( ( select ) => 
		select( OPTIONS_STORE_NAME ).isOptionsUpdating);
``` 
You can also return it as an object like before, that way it only updates once the updating changes.

Reply:
Tweaked in https://github.com/woocommerce/woocommerce-admin/pull/5775/commits/fcaeaaa619e07a9d2ac537035e01d7cb44cc4e6d.

Reply:
Nice looks good :)